scipy/special/_basic.py

```diff
@@ -2953,6 +2956,108 @@ def factorialk(n, k, exact=True):
         raise NotImplementedError
 
 
+def stirling2(N, K, exact=True):
+    """Generate Stirling number(s) of the second kind.
+
+        Stirling numbers of the second kind count the number
+        of non-empty subsets that can be formed by N elements
+        into K non-empty subsets.
+
+        This is often expressed as "N subset K".
```

Comment:
> Does this mean N⊂K? I've only seen the notations {NK}, S(N,K) and the other notations listed on [Wolfram Mathworld](https://mathworld.wolfram.com/StirlingNumberoftheSecondKind.html). 

Here I'm using the same line as Graham, Knuth, and Patashnik's concrete maths. Chapter 6, Page 258, where they describe how to say the notation verbally. Similar to how you would verbally read the notation 
```math
{n \choose{x}}
```
as 'n choose x', you read the notation 
```math 
{n \brace x}
```
as 'n subset x'. This may not be standard though so I can remove the line. That is less about the documentation and more about how to communicate the notation verbally.

Reply:
Huh, that's one of my favorite books but I didn't remember that detail.

![image](https://github.com/scipy/scipy/assets/1953382/64fc0c80-b44b-42e4-bbf4-7c9dba11a5dd)

*Concrete Mathematics* is where the brackets notation originates too. I wonder if n subset k didn't catch on or if I've just been unfortunate in never hearing it said out loud. I guess it depends upon the particular group whether ${n \brace k}$ is commonly expressed this way. 

Maybe write

`Graham, Knuth, and Patashnik recommend calling this quantity "N subset K" [1]`

and give the citation.

Reply:
For another data-point, Stanley uses the $S(N, K)$ notation and doesn't say "N subset K".

https://www.ms.uky.edu/~sohum/putnam/enu_comb_stanley.pdf

Reply:
yeah these are the two common ways of expressing the Stirling numbers of the second kind. I'm preferential to Knuth's notation-b/c of the analogies with Binomial coefficients-rather than the $S(N, K)$ but they're both commonly used. The bracket notation was (semi) popularized by Knuth but actually originates w/Jovan Karamata  (same as of the Karamata inequality).

here's a wiki ref: https://en.wikipedia.org/wiki/Jovan_Karamata#Legacy stating Karamata originated the curly braces notation.